diff -urN obconf-2.0.2/obconf.desktop obconf-2.0.2.new/obconf.desktop
--- obconf-2.0.2/obconf.desktop	2007-07-12 18:24:52.000000000 +0200
+++ obconf-2.0.2.new/obconf.desktop	2008-01-20 18:35:36.000000000 +0100
@@ -1,11 +1,12 @@
 [Desktop Entry]
-Encoding=UTF-8
 Type=Application
 Name=Openbox Configuration Manager
+Name[pl]=Zarządca ustawień Openboksa
 Comment=Openbox configuration manager
+Comment[pl]=Zarządca ustawień Openboksa
 Icon=openbox
 Exec=obconf %f
 Categories=Settings;DesktopSettings;GTK;
-MimeType=application/x-openbox-theme
+MimeType=application/x-openbox-theme;
 StartupNotify=true
 Terminal=false
